Add notice about template cookbook locations

emachnic opened the pull request:
When developing a custom resource, I found that the cookbook in which I was trying to call the resource could not find the templates and realized that I needed to specify the cookbook location where those templates live.

### Description

Adds a note about specifying the cookbook location in a custom resource template

### Definition of Done

### Issues Resolved

### Check List

- [ ] Spell Check
- [ ] Local build
- [ ] Examine the local build
- [ ] All tests pass

emachnic committed:
Add notice about template cookbook locations

When developing a custom resource, I found that the cookbook in which I was trying to call the resource could not find the templates and realized that I needed to specify the cookbook location where those templates live.

Signed-off-by: Evan Machnic <emachnic@gmail.com>

tas50 commented:
Thanks for adding the comment. I think we may shuffle this all around in the near future, but this is great for today and it'll remind us in the future to fully document how templates behave in a resource world.

IanMadd committed:
Merge branch 'master' into patch-1

tas50 commented:
I created a cookstyle issue for this since I think we can auto detect it: https://github.com/chef/cookstyle/issues/160

IanMadd committed:
Light re-edit for clarity

Signed-off-by: Ian Maddaus <imaddaus@chef.io>

Removed dtag because we already have too many and made some minor edits for clarity.

IanMadd reviewed (approved):
Thanks for the contribution @emachnic!

![tenor-202987062](https://user-images.githubusercontent.com/8138169/62874539-6b9a7100-bcef-11e9-8d8b-dc96a59cfff1.gif)
